### DISCUSSION TITLE
New Sonos integration does not see my Sonos boxes any more

### DISCUSSION THREAD

**Discussion by Simpelton** [music-assistant/support#3087] [Q&A] [answered]
With the recent update, my Sonos boxes are not detected any more. 

In the update notes, I noticed that the Sonos support had been revamped and that Sonos1`s are special. After the update, my Sonos boxes are no longer detected. I tried to find the update note and checked the Sonos player support docs but failed to relocate the update note. Checked the logs, but no luck. Restarted HA, no luck

My Sonos environment:

- MA worked nice before the update (now 2.3.1)
- Got Sonos2 players, one older box but that does support Sonos2 as well
- My Sonos boxes are **NOT** connected to the internet!

Due to recent events in the Sonos universe, I cut off internet access for my Sonos boxes. One reason is, I do not want Sonos to force me with updates into a lack of already existing (and payed for) functionality - their release notes suck big time, so there is no transparency what the next updates brings. And Sonos track record is awful in that respect. 
And I do not want to give the internet access to my bedroom - the Sonos web interface is only U:PW protected - no MFA last time I checked. No option to disable the web interface. I have no transparency nor trust into Sonos-IT and -Software development capabilities.

I guess the new MA Sonos integration is using a Sonos provided Web API to find my local devices? Any way how to manualy (via IP) register the devices and avoid cloud provider API`s? Particularly those of Sonos?

**Comment by OzGav** [comment]
@marcelveldt ?

**Comment by Simpelton** [comment, marked as answer]
Strange - update to the latest and greatest of HA, reboot, Sonos devices are found again.

**Reply by solarjunkev** [reply]
I am 100% with your Sonos rant mate
I have had Sonos for over a decade now with a mixture of S1's and S2's..
I HATE what Sonos did to their original customers with that move from S1>S2 
HATE THEM
Yes now we seem to have yet more problems?.. this time with our beloved open source HA - which are trying their best to beat this Sonos deplorable act...
I too lost my Sonos devices in HA/Music assistant with ther recent FW change...
All this constant change with Sonos remains a complete PITA... It makes me want to ditch Sonos entirely and move to something like Heos frankly..

1/ If I want to run S1 and S2's together.. what do I do now with the recent MA upgrade?
Do I run "Sonos S1" or "Sonos"? (or both?!) 

2/ I also like using the Sonos integration : and the Sonos Card. I note now a new message in the music assistant help section that recommends (under known issues) - “Using the Sonos HA Integration at the same time as the MA player provider may cause problems. Recommend only using the MA player provider”. What is this all about?

3/ What is the future for us users that still want to play their S1 devices? Is there a future??? Or should we ditch them? My how I loved that ZP100.. Sonos was never the same after they dropped support of that device and never upgraded it properly with a line in...

(I could rant on but ...you know how it is I am sure.. )..

Thanks

**Reply by OzGav** [reply]
1. You use the provider that is appropriate for your devices. You say you have a mixture of S1 and S2 so you would need both as I understand it.
2. It’s about not causing unnecessary problems. If you are using MA then you should not need any HA integrations for the supported player types. The MA players are optimised for music playback and if you use the integration then all of the MA players appear in HA
3. As above MA supports S1 devices

**Reply by solarjunkev** [reply]
Ok thanks - (it wasn't clear you needed _both_ Sonos and Sonos S1 running..)
Why can't music assistant get along with the Sonos integration?
What is all _that_ about? 
I'd like to run (and continue to try) both.. including the Sonos Card...
(I won't be the only one like this)
Cheers

**Reply by marcelveldt** [reply]
> I HATE what Sonos did to their original customers with that move from S1>S2

Really?! The original move was bad indeed but the current solution where even 15 year old speakers keep working with the S1 app is an example for the rest of the industry. Remember that the hardware in these speakers is at some point no longer capable of running newer features.

> All this constant change with Sonos remains a complete PITA... It makes me want to ditch Sonos entirely and move to something like Heos frankly..

And there you will run into the exact same problems. 

> If I want to run S1 and S2's together.. what do I do now with the recent MA upgrade?

You run both providers, just like you have 2 apps on your phone.

> What is the future for us users that still want to play their S1 devices? 

In case this wasn't already super clear: You just keep using the Sonos S1 provider. 


> Why can't music assistant get along with the Sonos integration?

That applies to the Sonos S1 provider only:
Because only one of them can run at the same time on the same host, otherwise the callback stops working. Also these speakers do not like too many requests from too many sources. If it works for you, then great, but you may run into issues where e.g. the track info is incorrect etc. simply because the callback of speaker updates stops working.

It is fine if you use the Sonos S2 provider in MA and the HA Sonos integration at the same time (although I wonder why really).

**Reply by solarjunkev** [reply]
Ok thanks for that info. I think I'll update the, to S2's Cheers
